### ISSUE TITLE
Client triggers pre-push hook when coming to foreground

### ISSUE THREAD

**Issue by pattogato** [desktop/desktop#9787]
### Describe the bug

I have pre-push hook set up, so it will do some formatting, etc.
I'm seeing this bug for the first time with this version, when I click on the Github app and it comes in foreground, it triggers the pre-push hook which is not expected.

### Version & OS
Version 2.5.0
Mac OS Catalina 10.15.2 (19C57)

### Steps to reproduce the behavior

1. Define a pre-push hook
2. Have some un-pushed commits
3. When the client comes in foreground, it triggers the pre-push hook (without pressing the push button)

### Expected behavior
Not triggering the hook unless I press the push button

### Logs

[2020-05-13.desktop.production.log.zip](https://github.com/desktop/desktop/files/4623496/2020-05-13.desktop.production.log.zip)

**Comment by tierninho** [comment]
@pattogato Thanks for filing this. I see two errors in the logs:

```
2020-05-13T15:30:37.268Z - error: [ui] `git rev-list --left-right --count aa0d989551df84122979845c11c3875a42b07c46...d2869a641b31a4898280c2e37352c105767a1a16 --` exited with an unexpected code: 128.
stderr:
fatal: Invalid symmetric difference expression aa0d989551df84122979845c11c3875a42b07c46...d2869a641b31a4898280c2e37352c105767a1a16
```

and

```
2020-05-13T15:30:08.205Z - info: [ui] proxy url not resolved, protocol not supported
```

We do have a [bug](https://github.com/desktop/desktop/issues/8847) related to git hook still pending a fix, but I am uncertain if they are related. 

cc @outofambit for any insight into this

**Comment by rafeca** [comment]
I think this is caused by by the checks that we've started doing in v2.5.0 for unpushed tags, that execute `git push --dry-run`.

We can solve this by adding the [`--no-verify`](https://git-scm.com/docs/git-push#Documentation/git-push.txt---no-verify) CLI argument to the push command.

If we find more edges cases with the current approach of detecting unpushed tags, we can investigate a simpler alternative (something that @niik recently mentioned me): we can store locally the tags that get created from Desktop and only push these (This way we won't need to do a network call). The main tradeoff with this simpler approach is that Desktop won't be aware of the tag-related actions that are performed outside of Desktop (create/delete/push tags), which can create a different set of edge cases.

**Comment by pattogato** [comment]
I'd do the `--no-verify` because in my understanding that verification is ment to be ran on new code. Does that make sense?

**Comment by hubgit** [comment]
I'm experiencing this problem as high CPU usage, caused by pre-push hooks being triggered in the background by GitHub Desktop. 

Adding `--no-verify` sounds like a good idea.

**Comment by steveward** [comment]
@david-shortman reported the same issue as @hubgit in #9789.

**Comment by cpcallen** [comment]
Unfortunately `--no-verify` will not prevent git from running ssh, which in turn for me (thanks to corporate security policy) causes a modal dialog to pop up, stealing focus from the foreground app to prompt me to touch my physical security key; see [my comment](https://github.com/desktop/desktop/issues/1128#issuecomment-628668920) on #1128.

**Comment by rafeca** [comment]
Thanks for your input @cpcallen, your case is definitely an edge case we didn't foresee when implemented tags support in v2.5.0 .

Desktop currently does a background network request to check whether any of the local tags have not been pushed to the remote repository, this is what's causing your popups.

We're studying alternative ways to perform such checks that don't need to do network requests, as the one mentioned in my previous comment, which would fix your issue:

> If we find more edges cases with the current approach of detecting unpushed tags, we can investigate a simpler alternative (something that @niik recently mentioned me): we can store locally the tags that get created from Desktop and only push these (This way we won't need to do a network call). The main tradeoff with this simpler approach is that Desktop won't be aware of the tag-related actions that are performed outside of Desktop (create/delete/push tags), which can create a different set of edge cases.

Apologies for the caused pain, we're working on a solution for this issue, and we'll keep this issue updated with any advancement.

**Comment by rafeca** [comment]
Thanks everyone for the report!

A fix for the pre=push hook issue have been merged and a new beta version of GitHub has been released with it. You can download the beta for [macOS](https://central.github.com/deployments/desktop/desktop/latest/darwin?env=beta) or [Windows](https://central.github.com/deployments/desktop/desktop/latest/win32?env=beta).

@cpcallen I've created https://github.com/desktop/desktop/issues/9800 to investigate solutions to solve your issue, please subscribe to it to get updates 😃
